### DISCUSSION TITLE
What does  `REQUIRED` track?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#182] [Q&A] [answered]
Should `REQUIRED` track the entire `caddr` `u` file path or should it only track the basename of the file path?  The 1st case would allow loading the same file name from different paths, while the 2nd consumes less resources.

**Comment by ruv** [comment, marked as answer]
`required` identifies a file by the provided name `(c-addr u)` at the least.  Whether it takes into account the full name (beyond `c-addr u`)  — depends on implementation. For a standard program it does not matter.

Yes, if `required` only tracked the base name (the name without the path), then two files with the same name form different directories could not be both loaded by `required`.  I think, this is not an option.
